src/nebari/schema.py

```diff
@@ -7,8 +7,12 @@
 from _nebari.version import __version__, rounded_ver_parse
 
 # Regex for suitable project names
-namestr_regex = r"^[A-Za-z][A-Za-z\-_]*[A-Za-z]$"
-letter_dash_underscore_pydantic = pydantic.constr(regex=namestr_regex)
+project_name_regex = r"^[A-Za-z][A-Za-z0-9\-_]{1,30}[A-Za-z]$"
```

Comment:
Trailing numbers should be valid as well I'd think? Is `foobar-123` disallowed intentionally?